flintrock/templates/spark/conf/spark-defaults.conf

```diff
@@ -0,0 +1 @@
+spark.jars.packages    org.apache.hadoop:hadoop-aws:{hadoop_version}
```

Comment:
I initially thought that I needed `aws-java-sdk`, but [a Spark committer suggested to me](https://github.com/apache/spark/pull/12004#issuecomment-274322741) that `hadoop-aws` will specify the correct dependency on `aws-java-sdk`. (Otherwise, you need to know what version of each library to pair with one another, which seems strange since you would expect that dependency to already be tracked somewhere.)

The clusters I'm launching seem to access S3 fine with just the `hadoop-aws` dependency.

When you get that exception, what exactly are you trying to run? Are you using an `s3a://` path? How are your AWS credentials specified?